src/components/RepoSelectDialog.tsx

```diff
@@ -35,9 +37,17 @@ const RepoSelectDialog: React.FC<Props> = ({ open, setOpen, defaultRepo }): Reac
   const apiStatus = useAppSelector(selectPreferenceAPIStatus);
 
   const [repoName, setDefaultRepoName] = React.useState<string>();
+  const [alertOpen, setDefaultAlertOpen] = React.useState<boolean>();
 
   const handleChange = (event: SelectChangeEvent<typeof repoName>) => {
     setDefaultRepoName(String(event.target.value) || '');
+
+    const visibility = repos.filter(r => r.name === String(event.target.value))[0]['visibility']
+    if(visibility === 'public'){
```

Comment:
Can we make this if inline. like `visibility === 'public' && setDefaultAlertOpen(true) || setDefaultAlertOpen(false);`